built_value_generator/lib/src/value_source_class.dart

```diff
@@ -1202,15 +1202,26 @@ abstract class ValueSourceClass
     if (comparedFields.isEmpty) {
       result.writeln('return ${name.hashCode};');
     } else {
-      result.writeln(
-          'return ${generateMemoizedHashCode ? '__hashCode ??= ' : ''}\$jf(');
-      result.writeln(r'$jc(' * comparedFields.length);
+      if (generateMemoizedHashCode) {
+        result.writeln('if (__hashCode != null) return __hashCode$notNull;');
+      }
+
       // Use a different seed for builders than for values, so they do not have
       // identical hashCodes if the values are identical.
-      result.writeln(forBuilder ? '1, ' : '0, ');
-      result.write(
-          comparedFields.map((field) => '${field.name}.hashCode').join('), '));
-      result.writeln('));');
+      final seed = forBuilder ? 1 : 0;
+      result.writeln('int hash  = $seed;');
```

Comment:
Please change to

`var _\$hash =`

--with `var` since there is no need to specify the type, and the `_\$` prefix to make it unlikely to clash with a field name, like we do for `_\$result` above.